mno_analysis_tools/window_of_downtime.py

```diff
@@ -0,0 +1,70 @@
+# Compute maximum window of time with 0 messages
+import json
+import pytz
+import iso8601
+from datetime import datetime
+from temba_client.v2 import Message
+from core_data_modules.logging import Logger
+
+
+def group(lst, n):
+    """group([0,3,4,10,2,3], 2) => [(0,3), (4,10), (2,3)]
+    
+    Group a list into consecutive n-tuples. Incomplete tuples are
+    discarded e.g.
+    
+    >>> group(range(10), 3)
+    [(0, 1, 2), (3, 4, 5), (6, 7, 8)]
+    """
+    return zip(*[lst[i::n] for i in range(n)])
+
+
+log = Logger(__name__)
+log.set_project_name("WindowOfDowntime")
+
+if __name__ == "__main__":
+    with open("data.txt", mode="r") as f:
+        file_content = f.readline()
+        output = json.loads(file_content)
+        f.close()
+
+    data = [Message.deserialize(val) for val in output]
+    period_with_msg = []
+    generated_outputs = []
+    generated_data = []
+    output_keys = ["traffic", "direction", "start", "end", "delta"]
+    for msg in data:
+        traffic = msg.urn.split(":")[0]
```

Comment:
I don't think this makes sense to be called traffic. It's really the method people are using to talk to us. Struggling for a good name, the best I can think of is "communication_medium"